ProjectPerson/src/Programmer.java

```diff
@@ -0,0 +1,40 @@
+public class Programmer extends Person {
+    private String companyName;
+
+    protected Programmer(String name, String designation, String companyName) {
+        super(name, designation);
+        this.companyName = companyName;
+    }
+
+    private String getCompanyName() {
+        return companyName;
+    }
+
+    private void coding() {
+        System.out.println(getName() + " is a senior developer , at " + getCompanyName());
+    }
+
+    @Override
+    protected void walk(int steps) {
```

Comment:
@maria-salmasian I think this functions can move to parent class without overriding in all child classes

Reply:
right! fixed it